pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -153,11 +190,20 @@ func (r *ReconcileCluster) newStatus(cluster *synv1alpha1.Cluster) error {
 	return nil
 }
 
+func (r *ReconcileCluster) getTenantCR(tenant types.NamespacedName) (*synv1alpha1.Tenant, error) {
+	tenantCR := &synv1alpha1.Tenant{}
+
+	err := r.client.Get(context.TODO(), tenant, tenantCR)
+	if err != nil {
```

Comment:
this `if` is unnecessary since we return `err` anyways, no?

Reply:
Tested that on playground: https://play.golang.org/p/RMqmbB4n9vD

I thought tenantCR will still remain `nil` in that case.